apps/web/src/components/AuthSteps.tsx

```diff
@@ -0,0 +1,188 @@
+import Link from "next/link";
+import { PropsWithChildren, useRef } from "react";
+import { useMutation } from "react-query";
+import { useAccount, useConnect } from "wagmi";
+import { useCeramic } from "../hooks/useCeramic";
+import { useLit } from "../hooks/useLit";
+import { identify, trackEvent } from "../lib/analytics";
+import { cn } from "../utils/classnames";
+
+type AuthStepProps = PropsWithChildren<{
+  title: string;
+  description: string;
+  completed: boolean;
+}>;
+
+function AuthStep({ title, description, completed, children }: AuthStepProps) {
+  return (
+    <li className="flex gap-4">
+      <span
+        className={cn(
+          " flex h-8 w-8 flex-shrink-0 items-center justify-center rounded-lg  bg-gray-100 text-center font-medium text-gray-500",
+          completed && "bg-green-400 text-white"
+        )}
+      >
+        {completed ? (
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            width="20"
+            height="20"
+            viewBox="0 0 24 24"
+            fill="none"
+            stroke="currentColor"
+            strokeWidth="2"
+            strokeLinecap="round"
+            strokeLinejoin="round"
+          >
+            <polyline points="20 6 9 17 4 12"></polyline>
+          </svg>
+        ) : (
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            width="24"
+            height="24"
+            viewBox="0 0 24 24"
+            fill="none"
+            stroke="currentColor"
+            strokeWidth="2"
+            strokeLinecap="round"
+            strokeLinejoin="round"
+          >
+            <circle cx="12" cy="12" r="10"></circle>
+          </svg>
+        )}
+      </span>
+      <div className="flex flex-col items-start gap-4">
+        <div className="gap-2">
+          <h2 className="text-lg font-medium text-gray-800">{title}</h2>
+          <p className="text-gray-500">{description}</p>
+        </div>
+        {children}
+      </div>
+    </li>
+  );
+}
+
+const fromAuthSteps = {
+  from: "Auth Steps Modal",
+};
+
+export function AuthSteps() {
+  const { isConnected } = useAccount();
+
+  // store the connected state in a ref to prevent it from being removed when connecting from the modal
+  const connectedRef = useRef(isConnected);
+
+  const { connect, isLoading, connectors } = useConnect({
+    onSuccess: (data) => {
+      identify(data.account);
+      trackEvent("Wallet Connected", {
```

Comment:
Can we centralise this so we don't get flaky data if we are missing to track this somewhere?